src/VisualStudio/Core/Def/Guids.cs

```diff
@@ -23,6 +26,11 @@ internal static class Guids
         public const string CSharpLibraryIdString = "58F1BAD0-2288-45b9-AC3A-D56398F7781D";
         public const string CSharpReplPackageIdString = "c5edd1ee-c43b-4360-9ce4-6b993ca12897";
 
+        /// <summary>
+        /// A <see cref="UIContext"/> that is set if there is a C# project in the <see cref="VisualStudioWorkspace"/>.
+        /// </summary>
+        public static readonly Guid CSharpProjectExistsInWorkspaceUIContext = new Guid("CA719A03-D55C-48F9-85DE-D934346E7F70");
```

Comment:
Shell currently has [`KnownUIContexts.CSharpProjectContext`](http://index/?query=CSharpProjectContext&rightProject=Microsoft.VisualStudio.Shell.15.0&file=KnownUIContexts.cs&rightSymbol=vfh7rco8yhep). However, it does not work for the scenario described in the PR description because it is only triggered when user clicks in the solution explorer (iirc).

We probably can't change that one for back compat reasons - but it would be good to add this context as `KnownUIContexts.CSharpProjectExistsInSolutionContext` and include doc comments to clarify the difference. This won't avoid the confusion completely, but at least the two will be displayed next to each other in intellisense and it will be easier for future consumers to figure out which one they really need...